src/util/std_expr.h

```diff
@@ -4491,13 +4491,16 @@ class infinity_exprt : public nullary_exprt
 };
 
 /// \brief A let expression
-class let_exprt:public exprt
+class let_exprt : public ternary_exprt
 {
 public:
-  let_exprt():exprt(ID_let)
+  let_exprt(
+    const symbol_exprt &symbol,
+    const exprt &value,
+    const exprt &where,
+    const typet &type)
+    : ternary_exprt(ID_let, symbol, value, where, type)
```

Comment:
`type` must always match `where.type()`, so you probably shouldn't take it as a constructor argument

Reply:
Thanks, done!